tests/integration_tests/performance/test_block_ab.py

```diff
@@ -240,3 +247,104 @@ def test_block_vhost_user_performance(
             **vm.dimensions,
         }
     )
+
+
+def _check_block_size(ssh_connection, dev_path, size):
+    """
+    Checks the size of the block device.
+    """
+    _, stdout, stderr = ssh_connection.run("blockdev --getsize64 {}".format(dev_path))
+    assert stderr == ""
+    assert stdout.strip() == str(size)
+
+
+@pytest.mark.nonci
+@pytest.mark.parametrize("vcpu_count", [1, 2], ids=["1vcpu", "2vcpu"])
+def test_vhost_user_block(microvm_factory, guest_kernel, rootfs, vcpu_count, metrics):
+    """
+    This test simply tries to boot a VM with vhost-user-block
+    as a root device and upload FirecrackerMetrics emitted by
+    the test to Cloudwatch.
+    """
+
+    orig_size = 10  # MB
+    new_sizes = [20, 10, 30]  # MB
+    vhost_user_socket = "/vub.socket"
+    mkfs_mount_cmd = "mkfs.ext4 /dev/vdb && mkdir -p /tmp/tmp && mount /dev/vdb /tmp/tmp && umount /tmp/tmp"
+
+    vm = microvm_factory.build(guest_kernel, rootfs, monitor_memory=False)
+    vm.spawn(log_level="Info")
+    vm.basic_config()
+    vm.add_net_iface()
+
+    # Add a block device to test resizing.
+    fs = drive_tools.FilesystemFile(size=orig_size)
+    _backend = spawn_vhost_user_backend(vm, fs.path, vhost_user_socket)
+    vm.add_vhost_user_drive("scratch", vhost_user_socket)
+    vm.start()
+
+    # Check that guest reports correct original size.
+    _check_block_size(vm.ssh, "/dev/vdb", orig_size * 1024 * 1024)
+
+    # Check that we can create a filesystem and mount it
+    ret, out, err = vm.ssh.run(mkfs_mount_cmd)
+    assert ret == 0, f"{ret}, {out}, {err}"
+
+    DIMENSIONS = {
+        "instance": global_props.instance,
+        "cpu_model": global_props.cpu_model,
+        "host_os": global_props.host_os,
+        "host_kernel": "linux-" + global_props.host_linux_version,
+        "guest_kernel": guest_kernel.name,
+        "vcpus": str(vcpu_count),
+    }
+
+    metrics.set_dimensions(
+        {
+            "performance_test": "test_vhost_user_block",
+            **DIMENSIONS,
+        }
+    )
+
+    for new_size in new_sizes:
+        # Instruct the backend to resize the device.
+        # It will both resize the file and update its device config.
+        resize_vhost_user_drive(vm, new_size)
+
+        # Instruct Firecracker to reread device config and notify
+        # the guest of a config change.
+        vm.patch_drive("scratch")
+
+        # Check that guest reports correct new size.
+        _check_block_size(vm.ssh, "/dev/vdb", new_size * 1024 * 1024)
```

Comment:
do we actually need this duplicated check here if it is not the purpose of the test?